### PR TITLE
feat: new column 'Time taken to Deliver' in sales order analysis

### LINKED ISSUE #28753: Include column for 'Days taken to deliver' in Sales Order Analysis Report
Currently, the Sales Order Analysis reflects the Delay in Days took them to deliver the Ordered Items, they want one more column to reflect how many days they take to completely deliver a Sales Order in order to increase the efficiency and get more productive accordingly.

### PR DESCRIPTION
**Pull request by ruthra-kumar**
### Current Report
_'Sales Order Analysis'_ helps in quickly identifying billed amount, pending amount, delay in days and more information. Along with this a new column is added

### New Column - 'Time taken to Deliver'
This denotes the no of days taken to completely deliver an item. This helps in identifying bottlenecks and get more efficient.

### Image
time_taken_to_deliver

**no-docs**

### CODE REVIEW

File: erpnext/selling/report/sales_order_analysis/sales_order_analysis.py [hunk 1]
```diff
@@ -259,6 +264,12 @@ def get_columns(filters):
 			"fieldname": "delay",
 			"fieldtype": "Data",
 			"width": 100
+		},
+		{
+			"label": _("Time Taken to Deliver"),
+			"fieldname": "time_taken_to_deliver",
+			"fieldtype": "Data",
```
Comment (nextchamp-saqib) [inline erpnext/selling/report/sales_order_analysis/sales_order_analysis.py]: Can we make this a Duration field? And store the seconds difference between `posting_date` & `transaction_date`?

Comment (ruthra-kumar) [reply erpnext/selling/report/sales_order_analysis/sales_order_analysis.py]: Changed to Duration

**Review by nextchamp-saqib — APPROVED** [review]

**Comment by marination** [comment]
@nextchamp-saqib @ruthra-kumar This report was introduced via https://github.com/frappe/erpnext/pull/22055 to replace "Ordered Items to be Billed" and "Ordered Items to be Delivered and Billed"
The right join to DN makes it only fetch those transaction that have DNs against them, it does not fetch SOs that are pending to Deliver and Bill
It could use a test too, given that this could have been identified by CI

Also this column is a bit confusing with the Delay column, this needs documentation too https://docs.erpnext.com/docs/v13/user/manual/en/selling/sales-analytics

**Comment by ruthra-kumar** [comment]
@marination thanks for reporting this. Will create a fix.

**Pull request merged**